docs/integrations/context-and-outputs.md

```diff
@@ -139,5 +141,630 @@ return_outputs(
+    ip='8.8.8.8',
+    asn='Google LLC',
+    dbot_score=dbot_score
+)
+
+command_results = CommandResults(
+    indicators=[ip]
+)
+
+return_results(command_results)
+```
+
+**Domain - Example**
+```python
+dbot_score = Common.DBotScore(
+    indicator='8.8.8.8',
+    indicator_type=DBotScoreType.IP,
+    integration_name='Virus Total',
+    score=Common.DBotScore.GOOD
+)
+
+domain = Common.Domain(
+    domain='google.com',
+    dns='ns3.google.com',
+    whois=WHOIS(
+        registrar_name='MarkMonitor Inc.',
+        registrar_abuse_email='abusecomplaints@markmonitor.com'
+    )
+)
+
+command_results = CommandResults(
+    indicators=[domain]
+)
+
+return_results(command_results)
+```
+
+### Examples:
+- [Return data (common case)](#return-data) 
+- [Return results with custom markdown](return-custom-markdown)
+- [Return potentially malicious file](return-file)
+- [Return non malicious file (e.g. reports)](return-entry-file)
+- [Return IP reputation](#return-ip-reputation)
+- [Return Domain reputation](#return-domain-reputation)
+- [Return URL reputation](#return-url-reputation)
+- [Return File reputation](#return-file-reputation)
+- [Return CVE reputation](#return-cve-reputation)
+
+#### <a name='return-data'></a> Return Data (common case)
+*Code*
+```python
+alerts = [
+    {
+        'id': 100,
+        'name': 'alert1'
+    },
+    {
+        'id': 200,
+        'name': 'alert2'
+    }
+]
+
+results = CommandResults(
+    outputs_prefix='PrismaCompute.Alert',
+    outputs_key_field='id',
+    outputs=alerts
+)
+return_results(results)
+
+```
+
+*YAML Definition*
+```yaml
+outputs:
+- contextPath: PrismaCloud.Alert.id
+  description: 'The alert id'
+  type: Number
+- contextPath: PrismaCloud.Alert.name
+  description: 'The alert name'
+  type: String
+```
+
+*Markdown*
+### Results
+|id|name|
+|---|---|
+| 100 | alert1 |
+| 200 | alert2 |
+
+*Context - The way it stored in incident context*
+```json
+{
+  "PrismaCompute": {
+    "Alert": [
+      {
+        "id": 100,
+        "name": "alert1"
+      },
+      {
+        "id": 200,
+        "name": "alert2"
+      }
+    ]
+  } 
+}
+```
+
+
+#### <a name='return-custom-markdown'></a> Return results with custom markdown
+```python
+alerts = [
+    {
+        'id': 100,
+        'name': 'alert1'
+    },
+    {
+        'id': 200,
+        'name': 'alert2'
+    }
+]
+
+markdown = '## This is header<br/><br/>'
+markdown += tableToMarkdown('Table title', alerts, headers=['id', 'name'])
+
+results = CommandResults(
+    readable_output=markdown,
+    outputs_prefix='PrismaCompute.Alert',
+    outputs_key_field='id',
+    outputs=alerts
+)
+return_results(results)
+```
+
+*Markdown*
+## This is header<br/><br/>
+### Table title
+|id|name|
+|---|---|
+| 100 | alert1 |
+| 200 | alert2 |
+
+#### <a name='return-file'></a> Return file (potentially malicious - e.g. email attachment)
+*Code*
+```python
+file_entry = fileResult(filename='file.txt', data='file content')
+demisto.results(file_entry)
```

Comment:
this is not related to this doc, but i think its a bit confusing that for file we dont use `return_results`
is that intended or we just did not make `return_results` support files?

Reply:
currently it is not supported. But in the future we will add support for that too.
Next I am thinking to add classes for DemistoIncident class, DemistoIndicator class, instead of passing dict to demisto.incidents(), demisto.indicators()